test/SecTester.Scan.Tests/ScanTests.cs

```diff
@@ -1,53 +1,228 @@
+
+  public override void Dispose()
+  {
+    _scans.ClearSubstitute();
+    _logger.ClearSubstitute();
+
+    base.Dispose();
+  }
+
+  [Fact]
+  public void Constructor_GivenNullId_ThrowError()
+  {
+    // act
+    Action act = () => new Scan(null!, _scans, _logger, new ScanOptions());
+
+    // assert
+    act.Should().Throw<ArgumentNullException>().WithMessage("*id*");
+  }
+
   [Fact]
   public void Constructor_GivenNullScans_ThrowError()
   {
     // act
-    var act = () => new Scan(null!, new ScanOptions());
+    Action act = () => new Scan(ScanId, null!, _logger, new ScanOptions());
 
     // assert
     act.Should().Throw<ArgumentNullException>().WithMessage("*scans*");
   }
 
+  [Fact]
+  public void Constructor_GivenNullLogger_ThrowError()
+  {
+    // act
+    Action act = () => new Scan(ScanId, _scans, null!, new ScanOptions());
+
+    // assert
+    act.Should().Throw<ArgumentNullException>().WithMessage("*logger*");
+  }
+
   [Fact]
   public void Constructor_GivenNullScanOptions_ThrowError()
+  {
+    // act
+    Action act = () => new Scan(ScanId, _scans, _logger, null!);
+
+    // assert
+    act.Should().Throw<ArgumentNullException>().WithMessage("*options*");
+  }
+
+  [Fact]
+  public async Task Active_DefaultValue_ReturnsTrue()
+  {
+    // act
+    await using var scan = new Scan(ScanId, _scans, _logger, new ScanOptions());
+
+    // assert
+    scan.Active.Should().BeTrue();
+  }
+
+  [Fact]
+  public async Task Done_DefaultValue_ReturnsTrue()
+  {
+    // act
+    await using var scan = new Scan(ScanId, _scans, _logger, new ScanOptions());
+
+    // assert
+    scan.Done.Should().BeFalse();
+  }
+
+  [Fact]
+  public async Task Status_ReturnsState()
   {
     // arrange
-    var scans = Substitute.For<Scans>();
+    var scanState = new ScanState(ScanStatus.Running);
+    await using var scan = new Scan(ScanId, _scans, _logger, new ScanOptions(PollingInterval: TimeSpan.Zero));
+
+    _scans.GetScan(ScanId).Returns(Task.FromResult(scanState));
 
     // act
-    var act = () => new Scan(scans, null!);
+    var result = await scan.Status().FirstAsync();
 
     // assert
-    act.Should().Throw<ArgumentNullException>().WithMessage("*options*");
+    result.Should().Be(scanState);
   }
 
   [Fact]
-  public void Active_DefaultValue_ReturnTrue()
+  public async Task Status_GivenCancelledToken_ThrowError()
   {
     // arrange
-    var scans = Substitute.For<Scans>();
+    using var cts = new CancellationTokenSource();
+    await using var scan = new Scan(ScanId, _scans, _logger, new ScanOptions(PollingInterval: TimeSpan.Zero));
+
+    cts.Cancel();
 
     // act
-    using var scan = new Scan(scans, new ScanOptions());
+    var act = async () => await scan.Status(cts.Token).FirstAsync();
 
     // assert
-    scan.Active.Should().BeTrue();
+    await act.Should().ThrowAsync<OperationCanceledException>();
   }
 
   [Fact]
-  public void Done_DefaultValue_ReturnTrue()
+  public async Task Status_SwitchingDoneState_StopsYield()
```

Comment:
The test case is not clear
```suggestion
  public async Task Status_FinalStateReached_StopsQueryingState()
```